oracles/oracle_ws_api.md

```diff
@@ -0,0 +1,427 @@
+   "payload":{"type":"query",
+              "oracle_id":"ok$3jzZyCLFtHVD7yVdEhGJFM3LjeXrKqWxnHbCYzhnrrR4DkdFtaJuxQvrR8VbbXExDPkCHFAei5q969JA6EayQpb8z5C3Mf"}
+  }
+
+Response:
+  {"origin":"oracle",
+   "action":"subscribe",
+   "payload":{"result":"ok",
+              "subscribed_to":{"type":"query",
+                               "oracle_id":"ok$3jzZyCLFtHVD7yVdEhGJFM3LjeXrKqWxnHbCYzhnrrR4DkdFtaJuxQvrR8VbbXExDPkCHFAei5q969JA6EayQpb8z5C3Mf"}}
+  }
+
+Event:
+  {"origin":"node",
+   "action":"new_oracle_query",
+   "payload":{"sender":"ak$3jzZyCLFtHVD7yVdEhGJFM3LjeXrKqWxnHbCYzhnrrR4DkdFtaJuxQvrR8VbbXExDPkCHFAei5q969JA6EayQpb8z5C3Mf",
+              "query":"How are you?",
+              "query_id":"oq$4RZoMEkm8QuuhJiiq53dd5pE4VstCthYRjBHgUKdhAhe7rLEr"}
+  }
+```
+
+### Register for query response events
+### Request
+ * **target:** `oracle`
+ * **action:** `subscribe`
+ * **payload:**
+
+  | Name | Type | Description | Required |
+  | ---- | ---- | ----------- | -------- |
+  | type | `response` | | Yes |
+  | query_id | string | | Yes |
+
+### Response
+ * **origin:** `oracle`
+ * **action:** `subscribe`
+ * **payload:**
+
+  | Name | Type | Description | Required |
+  | ---- | ---- | ----------- | -------- |
+  | result | string | `ok` or error reason | Yes |
+  | subscribed_to | object - response subscribe payload | | Yes on success |
+
+### Event
+ * **origin:** `node`
+ * **action:** `new_oracle_response`
+ * **payload:**
+
+  | Name | Type | Description | Required |
+  | ---- | ---- | ----------- | -------- |
+  | query_id | string | hash that identifies the query | Yes |
+  | query | string | | Yes |
+
+### Example
+```
+Request:
+  {"target":"oracle",
+   "action":"subscribe",
+   "payload":{"type":"response",
+              "query_id":"oq$4RZoMEkm8QuuhJiiq53dd5pE4VstCthYRjBHgUKdhAhe7rLEr"}
+  }
+
+Response:
+  {"origin":"oracle",
+   "action":"subscribe",
+   "payload":{"result":"ok",
+              "subscribed_to":{"type":"response",
+                               "query_id":"oq$4RZoMEkm8QuuhJiiq53dd5pE4VstCthYRjBHgUKdhAhe7rLEr"}}
+  }
+
+Event:
+  {"origin":"node",
+   "action":"new_oracle_response",
+   "payload":{"response":"I am fine, thanks!",
+              "query_id":"oq$4RZoMEkm8QuuhJiiq53dd5pE4VstCthYRjBHgUKdhAhe7rLEr"}
+  }
+```
+
+## Additional types
+### TTL
+
+| Name | Type | Description | Required |
+| ---- | ---- | ----------- | -------- |
+| type | string | `block` or `delta` | Yes |
+| value | number |  | Yes |
+
+### RelativeTTL
+
+| Name | Type | Description | Required |
+| ---- | ---- | ----------- | -------- |
+| type | `delta` |  | Yes |
+| value | number |  | Yes |
+
+
+**Current limitation:** Since there is currently no interface for offline signing
+(all signing is done by a node - and that node is associated with exactly one
+account) it is somewhat limited what you can do with oracles. I.e. in practice
+you need a node per account - obviously this is a limitation we should try to
+lift as soon as possible.
```

Comment:
This looks an action. Is this tracked somewhere and discussed?

Reply:
I created this task a while ago, which is related: https://www.pivotaltracker.com/story/show/153451598